SwiftIOSExperiments/Modules/Login/Interactor/LoginInteractor.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  LoginInteractor.swift
+//  SwiftIOSExperiments
+//
+//  Created by James Santos on 15/2/17.
+//  Copyright © 2017 James Santos. All rights reserved.
+//
+
+import UIKit
+
+class LoginInteractor: LoginUseCase {
+
+	weak var output: LoginInteractorOutput!
+
+	func authenticateUser(_ auth: Auth) {
+		AuthService.authenticate(auth.email, auth.password, completion: { [weak self] result in
+			DispatchQueue.main.async {
```

Comment:
```
typealias TransfertServiceResult = ServiceResultObject<Transfert, ServiceErrorType>
typealias PayeeListServiceResult = ServiceResultObject<Payees, ServiceErrorType>

protocol TransfertProvider {
  func requestTransfert(
    identifier: String, description: String, amount: Double, payee: Payee,
    completion: (EmptyResultCompletion)?)
  
  /// Get Benificiary list
  ///
  /// - parameter completion: completion closure
  func requestPayeeList(element: Payees?, setter: Payees? -> Void,
    cachePolicy: RequestCachePolicy, completion: (PayeeListServiceResult -> ())?)
  
  /// Validate an IBAN
  /// - parameter iban : The iban to validate
  /// - parameter completion : completion closure
  func validateIBAN(iban: String, completion: EmptyResultCompletion?)
  
  
  /// Add Payee
  /// - parameter Payee : payee to add
  /// - parameter completion : completion closure
  func addPayee(payee: Payee, completion: (ProfileEditionResult -> Void)?)
  
  /// Update Payee
  /// - parameter Payee : payee to update
  /// - parameter completion : completion closure
  func updatePayee(payee: Payee, completion: (ProfileEditionResult -> Void)?)
  
  /// Delete Payee
  /// - parameter Payee : payee to delete
  /// - parameter completion : completion closure
  func deletePayee(payee: Payee, completion: EmptyResultCompletion?)
  
}
```
with this protocol you can ensure your service interfaces. 
after that you create an implementation (alamofire/nsurlsession, etc) of this protocol.

```
class TransfertService: RemoteService, TransfertProvider
```
Generally you extends a mutualize services implementation to localize the alamofire call and implement the protocol for each usecase